### PR TITLE
Fix mypy typing errors

### PR DESCRIPTION
**Pull request by rth**
Fixes a few typing issues (detected with mypy).

```
mypy icubam --ignore-missing-imports
```
on master produced 33 errors,

<details>

```
icubam/time_utils.py:52: error: Incompatible return value type (got "Tuple[int, ...]", expected "Tuple[str, str]")
icubam/db/store.py:44: error: Incompatible types in assignment (expression has type "List[<nothing>]", target has type "Dict[Any, Any]")
icubam/db/store.py:48: error: "Dict[Any, Any]" has no attribute "append"
icubam/db/store.py:50: error: "Dict[Any, Any]" has no attribute "append"
icubam/db/store.py:57: error: Cannot assign to a type
icubam/db/store.py:61: error: "Type[Base]" has no attribute "metadata"
icubam/db/store.py:67: error: "Type[Base]" has no attribute "metadata"
icubam/db/store.py:73: error: "Type[Base]" has no attribute "metadata"
icubam/db/store.py:251: error: Incompatible return value type (got "Union[User, None, Any]", expected "bool")
icubam/db/store.py:347: error: Unexpected keyword argument "name" for "User"
icubam/db/store.py:347: error: Unexpected keyword argument "email" for "User"
icubam/db/store.py:347: error: Unexpected keyword argument "password_hash" for "User"
icubam/db/store.py:347: error: Unexpected keyword argument "is_admin" for "User"
icubam/db/store.py:383: error: Incompatible return value type (got "Optional[User]", expected "User")
icubam/db/store.py:471: error: Self argument missing for a non-static method (or an invalid type for self)
icubam/db/store.py:492: error: Incompatible return value type (got "ValueError", expected "int")
icubam/db/store.py:522: error: Self argument missing for a non-static method (or an invalid type for self)
icubam/db/store.py:524: error: Name 'self' is not defined
icubam/db/store.py:662: error: Item "None" of "Optional[User]" has no attribute "is_admin"
icubam/db/store.py:666: error: Item "None" of "Optional[User]" has no attribute "icus"
icubam/db/store.py:667: error: Item "None" of "Optional[User]" has no attribute "managed_icus"
scripts/populate_db.py:44: error: Name 'sys' is not defined
icubam/messaging/handlers/onoff.py:10: error: Incompatible types in assignment (expression has type "None", variable has type "int")
icubam/messaging/handlers/onoff.py:11: error: Incompatible types in assignment (expression has type "None", variable has type "List[int]")
icubam/messaging/handlers/onoff.py:13: error: Incompatible types in assignment (expression has type "None", variable has type "int")
icubam/backoffice/handlers/base.py:55: error: Argument 2 to "get_body_arguments" of "RequestHandler" has incompatible type "None"; expected "bool"
icubam/backoffice/handlers/base.py:57: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "List[str]")
icubam/backoffice/handlers/base.py:74: error: Incompatible return value type (got "Union[Dict[Any, Any], List[Any]]", expected "List[Any]")
icubam/messaging/client.py:25: error: Argument 4 to "OnOffRequest" has incompatible type "Optional[int]"; expected "int"
icubam/www/handlers/version.py:33: error: Incompatible return value type (got "Dict[str, str]", expected "List[Dict[Any, Any]]")
icubam/messaging/scheduler.py:15: error: Incompatible types in assignment (expression has type "None", variable has type "Message")
icubam/messaging/scheduler.py:62: error: Argument 3 to "ScheduledMessage" has incompatible type "float"; expected "int"
icubam/messaging/scheduler_test.py:141: error: Name 'test_schedule_all' already defined on line 127
Found 33 errors in 9 files (checked 69 source files)
```
</details>

now there are 14 left that are mostly due to mypy quirks and are not too relevant,

<details>

```
icubam/db/store.py:59: error: Cannot assign to a type
icubam/db/store.py:63: error: "Type[Base]" has no attribute "metadata"
icubam/db/store.py:69: error: "Type[Base]" has no attribute "metadata"
icubam/db/store.py:75: error: "Type[Base]" has no attribute "metadata"
icubam/db/store.py:349: error: Unexpected keyword argument "name" for "User"
icubam/db/store.py:349: error: Unexpected keyword argument "email" for "User"
icubam/db/store.py:349: error: Unexpected keyword argument "password_hash" for "User"
icubam/db/store.py:349: error: Unexpected keyword argument "is_admin" for "User"
icubam/db/store.py:665: error: Item "None" of "Optional[User]" has no attribute "is_admin"
icubam/db/store.py:669: error: Item "None" of "Optional[User]" has no attribute "icus"
icubam/db/store.py:670: error: Item "None" of "Optional[User]" has no attribute "managed_icus"
icubam/backoffice/handlers/base.py:55: error: Argument 2 to "get_body_arguments" of "RequestHandler" has incompatible type "None"; expected "bool"
icubam/backoffice/handlers/base.py:57: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "List[str]")
icubam/backoffice/handlers/base.py:74: error: Incompatible return value type (got "Union[Dict[Any, Any], List[Any]]", expected "List[Any]")
Found 14 errors in 2 files (checked 69 source files)
```
</details>

### CODE REVIEW

File: icubam/db/store.py [hunk 1]
```diff
@@ -659,6 +661,7 @@ def get_visible_bed_counts_for_user(self,
     """
     session = self._session()
     user = self._get_user(session, user_id)
+    # TODO: handle user=None here
     if force or user.is_admin:
```
Comment (rth) [inline icubam/db/store.py:665]: We need to handle the case of `user=None` here. I'm not sure if it should be an error or handled the same as admin? (_get_user can return None)

File: icubam/messaging/scheduler_test.py [hunk 2]
```diff
@@ -139,7 +139,7 @@ def test_schedule_all(self):
     self.assertGreater(len(self.scheduler.timeouts), len(names) + 1)
 
   @mock.patch('time.time', mock.MagicMock(return_value=fake_now))
-  def test_schedule_all(self):
+  def test_schedule_all_2(self):
```
Comment (olivierteboul) [inline icubam/messaging/scheduler_test.py:142]: this one is probably not useful

File: icubam/time_utils.py [hunk 3]
```diff
@@ -44,7 +44,7 @@ def localewise_time_ago(ts, locale, ts_reference=None):
   return template.format(delta=count, units=units)
 
 
-def parse_hour(hour, sep=':') -> Tuple[str, str]:
+def parse_hour(hour, sep=':') -> Tuple[Union[int, str], ...]:
```
Comment (olivierteboul) [inline icubam/time_utils.py:47]: when too complex, i d rather skip it

File: icubam/db/store.py [hunk 4]
```diff
@@ -34,7 +36,7 @@ def to_dict(self, max_depth=1, include_relationships=True) -> dict:
      max_depth: the maximum recursion depth.
     """
     columns = self.get_column_names(include_relationships=include_relationships)
-    result = {}
+    result: Dict[str, Any] = {}
```
Comment (olivierteboul) [inline icubam/db/store.py:39]: nice but maybe a bit forced here. i would simply remove it

File: icubam/db/store.py [hunk 5]
```diff
@@ -489,7 +491,7 @@ def add_region(self, admin_user_id: int, region: Region) -> int:
       ID of the region.
     """
     if region.region_id:
-      return ValueError("ID of a new region should not be set.")
+      raise ValueError("ID of a new region should not be set.")
```
Comment (olivierteboul) [inline icubam/db/store.py:494]: good catch

**Comment by olivierteboul** [comment]
But thanks overall !

File: icubam/messaging/scheduler_test.py (hunk 2 above)
Comment (rth) [reply icubam/messaging/scheduler_test.py:142]: This one is actual critical: there are two test methods `test_schedule_all` now in this file and only one is being run by pytest. Granted this name could be improved, any suggestions?

File: icubam/time_utils.py (hunk 3 above)
Comment (rth) [reply icubam/time_utils.py]: Reverted. Complex types often mean that the API has an issue. Here, this function returns a `Tuple(int, int)` or `("", "")` I guess could be done better.

**Comment by rth** [comment]
Thanks @olivierteboul ! Comments addressed.

**Comment by rth** [comment]
Any suggestions about https://github.com/icubam/icubam/pull/167#discussion_r403485089  ? Not handling None there looks like a bug.

File: icubam/db/store.py (hunk 1 above)
Comment (sgirgin) [reply icubam/db/store.py:666]: It should be an error I think. A missing user should not behave like an admin.

**Comment by rth** [comment]
The `SchedulerTestCase.test_schedule_all` currently fails following changes introduced in https://github.com/icubam/icubam/pull/165. We need to merge this soon because that test is currently silently ignored by pytest on master @olivierteboul (so there is no point changing it in other PRs until this is fixed).

**Comment by rth** [comment]
> following changes introduced in #165. 

No my mistake, not that PR, but in any case this test was passing yesterday, and it doesn't today with no changes in this PR.

File: icubam/db/store.py (hunk 1 above)
Comment (rth) [reply icubam/db/store.py:666]: Thanks @sgirgin I opened https://github.com/icubam/icubam/issues/180 to address this in a follow up PR.

Comment (sgirgin) [reply icubam/db/store.py:666]: Great, thank you!

**Comment by rth** [comment]
Thanks!

**Pull request merged**